### PR TITLE
 [IMP] mail: introduce MessageListViewMessageViewItem 

### PR DESCRIPTION
**Pull request by seb-odoo**
This allows to remove the AND/OR mix from message view identifying fields.

Part of task-2741982

### CODE REVIEW

File: addons/mail/static/src/models/message_list_view_message_view_item.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { insertAndReplace } from '@mail/model/model_field_command';
+
+/**
+ * Models a relation between a message list view and a message view where
+ * message is used as iterating field.
+ */
+registerModel({
+    name: 'MessageListViewMessageViewItem',
```
Comment (seb-odoo) [inline addons/mail/static/src/models/message_list_view_message_view_item.js:12]: Other name suggestion:
- MessageListViewMessageView
- MessageListViewMessageViewByMessage
- MessageListViewMessageViewPerMessage

**Review by alexkuhn — APPROVED** [review]

**Comment by alexkuhn** [comment]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo rebase-ff

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**